falcon/testing/client.py

```diff
@@ -528,6 +534,15 @@ def simulate_request(
             overrides `body` and sets the Content-Type header to
             ``'application/json'``, overriding any value specified by either
             the `content_type` or `headers` arguments.
+        files(dict): same as the  files parameter in requests,
+             dictionary of ``'name': file-like-objects`` (or ``{'name': file-tuple}``)
+             for multipart encoding upload.
+            ``file-tuple``: can be a 2-tuple ``('filename', fileobj)``,
+                3-tuple ``('filename', fileobj, 'content_type')``
+                or a 4-tuple ``('filename', fileobj, 'content_type', custom_headers)``,
```

Comment:
Not sure if custom headers are really a must to support in the first iteration :thinking: Falcon doesn't really support any of these since they are forbidden by the latest RFCs.

Reply:
I used Content-Disposition for the nested files, just because you had it in test_media_multipart, but I'll remove them.